asmcomp/cmm.mli

```diff
@@ -134,10 +134,10 @@ and expression =
   | Clet of Ident.t * expression * expression
   | Cassign of Ident.t * expression
   | Ctuple of expression list
-  | Cop of operation * expression list
+  | Cop of operation * expression list * Debuginfo.t
   | Csequence of expression * expression
   | Cifthenelse of expression * expression * expression
-  | Cswitch of expression * int array * expression array
+  | Cswitch of expression * int array * expression array * Debuginfo.t
```

Comment:
Not quite. In that future, I would expect the optimisation of a local reference to produce both `Cassign` and some "phantom" operation like the "phantom" let in Mark's other PR. The phantom version would have the debuginfo and the `Cassign` wouldn't.

Well actually by the time DWARF 6 is out I would hope we had removed `Cassign` in favour of SSA -- but the "phantom" reference mutation would still exist and would still have debuginfo attached.